pyomo/opt/base/problem.py

```diff
@@ -44,3 +50,12 @@ def __enter__(self):
     def __exit__(self, t, v, traceback):
         pass
 
+
+class BranchDirection(object):
```

Comment:
Open question: is this the best place for this enum-like object?

Reply:
I wondered the same thing, but I'm not sure of a better place for it. Gurobi has the same options (up, down, and default).

Reply:
Yeah, this was the only place general enough that I could think of as there's nothing else as suitable. We could always introduce a new module for this though?